nsv13/code/modules/mob/living/carbon/human/species_types/dwarves.dm

```diff
@@ -0,0 +1,254 @@
+
+		for(var/key in dwarf_words) //Theres like 1459 words or something man.
+			var/value = dwarf_words[key] //Thus they will always be in character.
+			if(islist(value)) //Whether they like it or not.
+				value = pick(value) //This could be drastically reduced if needed though.
+
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
+			message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
+
+		if(prob(3))
+			message += pick(" By Armok!")
+	speech_args[SPEECH_MESSAGE] = trim(message)
+
+//This mostly exists because my testdwarf's liver died while trying to also not die due to no alcohol.
+/obj/item/organ/liver/dwarf
+	name = "dwarf liver"
+	icon_state = "liver"
+	desc = "A dwarven liver, theres something magical about seeing one of these up close."
+	alcohol_tolerance = 0 //dwarves really shouldn't be dying to alcohol.
+	toxTolerance = 5 //Shrugs off 5 units of toxins damage.
+	maxHealth = 150 //More health than the average liver, as you aren't going to be replacing this.
+	//If it does need replaced with a standard human liver, prepare for hell.
+
+//alcohol gland
+/obj/item/organ/dwarfgland
+	name = "dwarf alcohol gland"
+	icon_state = "plasma" //Yes this is a actual icon in icons/obj/surgery.dmi
+	desc = "A genetically engineered gland which is hopefully a step forward for humanity."
+	w_class = WEIGHT_CLASS_NORMAL
+	var/stored_alcohol = 250 //They start with 250 units, that ticks down and eventaully bad effects occur
+	var/max_alcohol = 500 //Max they can attain, easier than you think to OD on alcohol.
+	var/heal_rate = 1.5 //The rate they heal damages over 400 alcohol stored. Default is 0.5 so we times 3 since 3 seconds.
+	var/alcohol_rate = 30 //Its times 0.025 making it tick down by .25 per loop. This is .75. Default is 10, so we times 3 since 3 seconds.
+	var/run_eth_check_cycle = TRUE //We start the ethanol reagent mob search cycle
+	var/run_filth_check_cycle = TRUE //We start the area scan filth cycle
+	var/filth_cycle_interval = 5 SECONDS //5 second delay before each forview scan
+	var/eth_cycle_interval = 3 SECONDS //3 second delay before each eth reagent check
+
+/obj/item/organ/dwarfgland/prepare_eat()
+	var/obj/S = ..()
+	S.reagents.add_reagent("ethanol", stored_alcohol/10)
+	return S
+
+/obj/item/organ/dwarfgland/on_life() //Primary loop to hook into to start delayed loops for other loops..
+	if(run_eth_check_cycle == TRUE) //If first run_eth_check_cycle is equal to TRUE.
+		dwarf_eth_cycle() //Call Dwarf_eth_cycle - aka ethanol.
+		run_eth_check_cycle = FALSE //Then we set it to FALSE, proc sets it back to true on the timer.	
+	if(run_filth_check_cycle == TRUE) //If first_filth_check_cycle is equal to TRUE.
+		dwarf_filth_cycle() //Call Dwarf_filth_cycle.
+		run_filth_check_cycle = FALSE //Then we set it to FALSE, proc sets it back to true on the timer.
```

Comment:
You don't need to compare them as the value is 1.
`if(run_filth_check_cycle)`

Reply:
Ah, my bad I accidentally committed it since I was afraid of a power outage since Im in WV, I devised a new ticker after that iteration that checks and stops the loops if the mob is dead, and also handles each interval separately. So the for_view will only fire every 3 ticks, which should be around 6 seconds, and the mob reagent check will fire every tick which is 2 seconds, with adjustments from On_life's usual slowdowns affecting it etc. (My apologies If im incorrect, I was told a tick is 2 seconds and I stop-watched it and it seems to be close to that)